docs/design/environment/enviroment_management.md

```diff
@@ -0,0 +1,153 @@
+
+The upgrade defines the role of platform engineers and the concepts of **Module** and **Environment**. The Module is a building block enabling the delivery of a set of resources from **Schema Configuration** and **Environment Configuration**. While the Environment is a collection of application-independent configurations, and an environment configuration is composed of a set of module inputs, which forms the entire module inputs combined with schema configuration.
+
+<br />![workflow](workflow.png)
+
+## Goals and Non-Goals
+
+### Goals
+
+1. Give the definition of environment; 
+2. Define the writing format of environment configurations;
+3. Provide the commands to manage environments;
+4. Give the integration scheme with AppConfiguration when running `kusion build`.
+
+### Non-Goals
+
+1. Only support local file based environment management, the support of other storage backends is not the goal of version 0.10.0;
+2. Only consider using Kusion in CLI format, the usage of service layer is not considered.
+
+## Proposal
+
+### Definition
+
+Environment is a logical concept representing a target that a stack will be deployed to, which contains a set of configurations, kubeconfig and provider authentication information that can be reused by multiple stacks. We recommend environments grouped by SDLC phases or cloud vendors. For example, it could be named as dev, staging, and prod or cloud vendors such as AWS, Azure, and Aliyun.
+
+At the product level, Kusion does not support multiple clouds or multiple regions in an environment. Although users can define an artful module which builds a schema into multiple resources in multiple clouds or regions. Kusion cannot forbid such modules, but doesn't recommend it and will not provide supportive technical means of multiple clouds or region in an environment. If a platform team needs to manage multiple clouds or regions, just create multiple environments. 
+
+Environment config are kept by yaml file when stored locally, and only includes a set of module inputs for now. But it may include other data, such as backend, which is not appropriate to set in the configuration file. There are two reasons: 1. backend info is a global config rather than environment config; 2. backend info includes sensitive data, such as database password.
```

Comment:
Based on my experience, backend configurations typically vary across different environments. For example, the OSS path for prod is not the same as dev. Hence, it is reasonable to set up backend configs in the environment config.